fivox/cuda/simpleLFP.cu

```diff
@@ -0,0 +1,128 @@
+
+/* Copyright (c) 2014-2016, EPFL/Blue Brain Project
+ *                          Jafet.VillafrancaDiaz@epfl.ch
+ *
+ * This file is part of Fivox <https://github.com/BlueBrain/Fivox>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "simpleLFP.h"
+
+namespace cuda
+{
+__global__ void kernelLFP( const float* __restrict__ eventsX,
+                           const float* __restrict__ eventsY,
+                           const float* __restrict__ eventsZ,
+                           const float* __restrict__ radii,
+                           const float* __restrict__ values,
+                           const Parameters& params, const VolumeInfo& volInfo,
+                           float* __restrict__ output )
+{
+    // 1D grid of 1D blocks
+    const int threadId = blockIdx.x * blockDim.x + threadIdx.x;
+
+    const int xIndex = threadId % volInfo.dimensions.x;
+    const int yIndex = ( threadId / volInfo.dimensions.x )
+                       % volInfo.dimensions.y;
+    const int zIndex = (( threadId / volInfo.dimensions.x )
+                        / volInfo.dimensions.y ) % volInfo.dimensions.z;
```

Comment:
I think that if I use a 3D block with a straightforward 1-to-1 mapping between threads and voxels, the Z dimension is very limited (64 in the Tesla K20m). I'm sure it can be improved, and it will, but that was not the goal of this PR anyway, but just to have a first, basic implementation.

The bounds check I accidentally removed, I will put it back.